lnrpc/routerrpc/router_server.go

```diff
@@ -280,7 +280,18 @@ func (s *Server) SendToRoute(ctx context.Context,
 		return nil, err
 	}
 
-	preimage, err := s.cfg.Router.SendToRoute(hash, route)
+	total := lnwire.MilliSatoshi(req.TotalAmtMsat)
+
+	var addr lntypes.Hash
+	if len(req.PaymentAddr) > 0 {
+		addr, err = lntypes.MakeHash(req.PaymentAddr)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse "+
+				"payment_addr: %v", err)
+		}
+	}
+
+	preimage, err := s.cfg.Router.SendToRouteMPP(hash, route, total, addr)
```

Comment:
So then this assumes the legacy (current) payment type won't be used via this RPC method?